### PR TITLE
feat: report deployments to datadog

### PR DESCRIPTION
**Pull request by sven-urbanski-freiheit-com**
Reference: SRX-PLTW20

### CODE REVIEW

File: services/cd-service/pkg/repository/transformer_test.go [hunk 1]
```diff
@@ -3998,6 +4008,94 @@ func TestSendRegularlyDatadogMetrics(t *testing.T) {
 	}
 }
 
+type MockClient struct {
+	events []*statsd.Event
```
Comment (hannesg) [inline services/cd-service/pkg/repository/transformer_test.go:4012]: You can remove most stub implementations by adding `statsd.ClientInterface` here.

File: services/cd-service/pkg/repository/transformer.go [hunk 2]
```diff
@@ -121,11 +122,37 @@ func UpdateDatadogMetrics(state *State) error {
...
+					return fmt.Sprintf(" for team %s", oneChange.Team)
+				}
+				return ""
+			}()
+			event := statsd.Event{
+				Title:     "Kuberpult app deployed",
+				Text:      fmt.Sprintf("Kuberpult has deployed %s to %s%s", oneChange.App, oneChange.Env, teamMessage),
+				Timestamp: now,
+				Tags: []string{
+					"app:" + oneChange.App,
```
Comment (hannesg) [inline services/cd-service/pkg/repository/transformer.go:148]: Maybe rather `kuberpult.application`, `kuberpult.environment` and `kuberpult.team`? The tags are very generic at the moment and datadog is weird when the tags collide.

File: services/cd-service/pkg/repository/repository.go [hunk 3]
```diff
@@ -562,6 +562,11 @@ func (r *repository) ProcessQueueOnce(ctx context.Context, e element, callback P
 	span, ctx := tracer.StartSpanFromContext(e.ctx, "PostPush")
 	defer span.Finish()
 
+	ddError := UpdateDatadogMetrics(r.State(), changes)
```
Comment (hannesg) [inline services/cd-service/pkg/repository/repository.go:565]: This is the most performance sensitive part of the code that we have. Is that really safe to do? Maybe rather add dedicated Background task for sending and just put this on a bounded queue?

File: services/cd-service/pkg/repository/transformer.go (hunk 2 above)
Comment (bjoern-michaelsen-freiheit) [reply services/cd-service/pkg/repository/transformer.go:148]: probably better, yes.

Comment (sven-urbanski-freiheit-com) [reply services/cd-service/pkg/repository/transformer.go]: done

File: services/cd-service/pkg/repository/transformer_test.go (hunk 1 above)
Comment (sven-urbanski-freiheit-com) [reply services/cd-service/pkg/repository/transformer_test.go:4012]: done

File: services/cd-service/pkg/repository/repository.go (hunk 3 above)
Comment (sven-urbanski-freiheit-com) [reply services/cd-service/pkg/repository/repository.go]: as discussed, I added a feature toggle.

**Review by hannesg — APPROVED** [review]

**Pull request merged**